scripts/dts/edtlib.py

```diff
@@ -550,6 +550,15 @@ def _init_prop(self, name, options):
             # 'category: optional' property that wasn't there
             return
 
+        enum = options.get("enum")
+        if enum and not isinstance(enum, list):
```

Comment:
Thinking about it, we should have this in `_check_binding()`. As it is now, the check is skipped if we return above, and the whole binding should always be checked if possible.

Reply:
Moved into _check_binding()